src/common/guac_rect.h

```diff
@@ -81,5 +103,40 @@ void guac_common_rect_extend(guac_common_rect* rect, const guac_common_rect* min
  */
 void guac_common_rect_constrain(guac_common_rect* rect, const guac_common_rect* max);
 
+/**
+ * Check whether a rectangle intersects another
+ *
+ * @param rect
+ *     Rectangle to check for intersection
+ *
+ * @param master
+ *     The master rectangle
+ *
+ * @return
+ *     -1 if no intersection, 1 if first rect is completely inside the
+ *     master, 0 if partial intersection
+ */
+int guac_common_rect_intersects(guac_common_rect* rect,
```

Comment:
Please ensure the function and parameter descriptions end with the requisite period (see above).